std-bits/base/src/main/java/org/enso/base/Array_Builder.java

```diff
@@ -37,21 +41,94 @@ public boolean isEmpty() {
    * @param e the element to add
    */
   public void add(T e) {
-    builder.add(e);
+    if (objectArray != null) {
+        if (size == objectArray.length) {
+            objectArray = Arrays.copyOf(objectArray, size * 2);
+        }
+        objectArray[size++] = e;
+    } else if (primitiveArray instanceof long[] longArray) {
+        if (e instanceof Long l) {
+            if (size == longArray.length) {
+                primitiveArray = longArray = Arrays.copyOf(longArray, size * 2);
+            }
+            longArray[size++] = l;
+        } else {
+            objectArray = new Object[longArray.length];
+            for (int i = 0; i < longArray.length; i++) {
+                objectArray[i] = longArray[i];
+            }
```

Comment:
```suggestion
            for (int i = 0; i < size; i++) {
                objectArray[i] = longArray[i];
            }
```

Why do we copy the whole capacity? The elements beyond the current `size` should not matter to us, right?

In particular, I assume they will be initialized to `(long) 0` so they will create `(Long) 0` instances in our Object array, but having the default `null`s there for the uninitialized part seems to make much more sense.